futures-util/src/stream/futures_unordered/node.rs

```diff
@@ -146,3 +167,24 @@ unsafe impl<T> UnsafeWake for ArcNode<T> {
         Node::wake(node)
     }
 }
+
+unsafe impl<T> UnsafeWake for ArcNodeUnowned<T> {
+    #[inline]
+    unsafe fn clone_raw(&self) -> Waker {
```

Comment:
This needs to actually clone the underlying `Arc`, since the downstream future being polled could call `cx.local_waker().clone()` and then store the value indefinitely.

Reply:
Yes. It clones the `Arc` with `Node::waker(node)`.

Reply:
Oh gosh that was not obvious to me at all, but I see it now. Can you leave a comment?

Reply:
And the one created by `Node::waker` here will drop itself because it has been cast to a `*const Arc<Node<T>>`, rather than a `*const ArcNodeUnowned<T>`?

Reply:
Added the comment